### PR TITLE
Adding Timeout handling in question

### PR DESCRIPTION
**Pull request by terry182**
This should solve #9.

If there is any problem please let me know in order to fix it asap

### CODE REVIEW

**Comment by RonakR** [comment]
Hey @terry182, I just ran this a couple times, and while the timeout is woking fine, the question isn't being sent when you call `!trivia`. Instead, it is being sent with the timeout text. 
Also, there is a peculiar interaction when you set the `duration` higher and try to type the text after `!trivia`. It doesn't pop up in the bot's logs until after the timeout. 

Here are the logs for what's happening: 
```
2016-10-15 02:14:14,414 - triviaBot - INFO - [IN] [#lolbottest] <BaronVonRatata!~textual@c-24-11-175-212.hsd1.sc.comcast.net> !trivia
2016-10-15 02:14:14,416 - triviaBot - INFO - [EVT] event 'this is just fantasy' registered to <bound method Question.solve_question of <mod_trivia.Question instance at 0x103396518>> in #lolbottest
2016-10-15 02:14:14,416 - triviaBot - INFO - [OUT] [#lolbottest] Is this real life?
2016-10-15 02:14:19,417 - triviaBot - INFO - [EVT] event 'this is just fantasy' deleted
2016-10-15 02:14:19,417 - triviaBot - INFO - [OUT] [#lolbottest] Time is up! The correct answer is 'this is just fantasy'
```
According to the logs everything is working fine But the times in this image says otherwise: 
image

**Comment by terry182** [comment]
@RonakR I have done having the question pop up right after the command but now the timeout message is really delayed like

screen shot 2016-10-15 at 3 19 15 pm

but the time stamp is 
```
Adurna|Trivia     Is this real life?  15:12:41
Adurna|Trivia   Time is up! The correct answer is 'this is just fantasy'  15:13:24
```

I guess this behaviour is tend to be the problem of the beaviour send_msg

File: triviabot/bot.py [hunk 1]
```diff
@@ -30,6 +30,9 @@ def consume(self, bot, user, channel):
             self.consumed = True
             self.callback(bot, user, channel)
 
+        def is_consumed(self):
```
Comment (SaschaMann) [inline triviabot/bot.py]: this isn't Java, getters & setters aren't needed

Comment (terry182) [inline triviabot/bot.py]: Got it.

**Comment by SaschaMann** [comment]
It seems to work for me (after moving things a few lines down to resolve the merge conflict because I changed mod_trivia earlier). I can't replicate the behaviour you mentioned above.

**Comment by SaschaMann** [comment]
Just after typing that I saw your update, I'll let @RonakR test it, since he seemed to run into issues before that I didn't have.

**Comment by terry182** [comment]
@SaschaMann I just submitted a commit some minutes before. You mean the working one is the one with `thread.Timer` or the new one

**Comment by SaschaMann** [comment]
For me the one with `threading.Timer()` worked, I haven't tried out the new one yet.

File: triviabot/mod_trivia.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
+from twisted.words.protocols import irc
```
Comment (SaschaMann) [inline triviabot/mod_trivia.py]: This import doesn't appear to be used anywhere?

**Comment by SaschaMann** [comment]
Tested it and ran fine, merged into develop. Thanks for your work!

**Pull request merged**